### PR TITLE
Alerting: Change error log to warning and apply correct format when updating historic config

### PR DESCRIPTION
**Pull request by santihernandezc**
The call to `st.Logger.Error()` was using an incorrect format.
I also changed it to a warning after realizing that some harmless situations could lead to this log line (e.g. starting with an empty history table).

### CODE REVIEW

File: pkg/services/ngalert/store/alertmanager.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (st *DBstore) MarkConfigurationAsApplied(ctx context.Context, cmd *models.M
 		}
 
 		if rowsAffected != 1 {
-			st.Logger.Error("update statement affected %d alert configuration history records", rowsAffected)
+			st.Logger.Warn("Alert configuration history updated", "rows", rowsAffected)
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/store/alertmanager.go:157]:
```suggestion
			st.Logger.Warn("Unexpected number of rows updating alert configuration history", "rows", rowsAffected)
```

**Review by grobinson-grafana — COMMENTED** [review]
I would change it to something like this. That is, to add some kind of explanation that it is unexpected.

File: pkg/services/ngalert/store/alertmanager.go (hunk 1 above)
Comment (alexweav) [inline pkg/services/ngalert/store/alertmanager.go]:
```suggestion
			st.Logger.Warn("No alertmanager configs were found to mark as applied", "org", cmd.OrgID, "hash", cmd.ConfigurationHash)
```
Due to the Limit(1) it's only possible to hit this when `rowsAffected` is `0`, so the rows attribute isn't very useful. Including the hash can help when debugging.

Comment (santihernandezc) [reply pkg/services/ngalert/store/alertmanager.go]: Or that all configs are marked, that's why I didn't want to specify a certain scenario.

Comment (alexweav) [reply pkg/services/ngalert/store/alertmanager.go]: @grobinson-grafana's wording above is also good, I opened my browser prior to that commit so I didn't see it

**Review by grobinson-grafana — APPROVED** [review]
LGTM!

> ```WARN [02-06|18:40:41] Unexpected number of rows updating alert configuration history logger=ngalert rows=0 org=1 hash=not-yet-calculated```

**Review by alexweav — APPROVED** [review]
LGTM

**Pull request merged**